align/schema/constraint.py

```diff
@@ -520,38 +524,60 @@ class DoNotIdentify(SoftConstraint):
 class SymmetricBlocks(SoftConstraint):
     pairs: List[List[str]]
     direction: Literal['H', 'V']
+    mirror: Optional[bool] = True
+
     def check(self, checker):
         '''
         X = Align(2, 3, 'h_center')
         Y = Align(4, 5, 'h_center')
         Align(1, X, Y, 6, 'center')
 
         '''
-        #TODO:function to check before adding, right now it adds and then check
+        # TODO:function to check before adding, right now it adds and then check
         assert all(len(pair) for pair in self.pairs) >= 1, 'Must contain at least one instance'
         assert all(len(pair) for pair in self.pairs) <= 2, 'Must contain at most two instances'
         if not hasattr(self.parent.parent, 'elements'):
             # PnR stage VerilogJsonModule
+            logger.debug(f"Skipping constraint checking 1 {self.parent} ")
+            for pair in self.pairs:
+                if len(pair) == 2:
+                    bvars = checker.iter_bbox_vars(pair)
+                    for b1, b2 in itertools.pairwise(bvars):
+                        if self.direction == 'V':
+                            pd = 'sy'
+                            od = 'sx'
+                        else:
+                            pd = 'sx'
+                            od = 'sy'
+                        checker.append(getattr(b1, pd) == getattr(b2, pd))
+                        if self.mirror:
+                            checker.append(getattr(b1, od) != getattr(b2, od))
+                        else:
+                            checker.append(getattr(b1, od) == getattr(b2, od))
             return
-        if len(self.parent.parent.elements)==0:
-            #skips the check while reading user constraints
+        if len(self.parent.parent.elements) == 0:
+            # skips the check while reading user constraints
+            logger.debug(f"Skipping constraint checking 2 {self.parent} ")
```

Comment:
I think the way constraint read is built right now is we read in the JSON then append one by one. We should discuss this so that we can check the designer input constraints. I will set up a meeting for this.